.github/workflows/docker-pre-release.yml

```diff
@@ -0,0 +1,77 @@
+---
+name: Create pre-release and push Docker image
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'api/**'
+      - 'client/**'
+      - 'Dockerfile'
+      - '.dockerignore'
+      - '.github/workflows/docker-release.yml'
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+jobs:
+  create_new_release:
+    permissions:
+      contents: write
+      pull-requests: write
+      packages: write
+    runs-on: ubuntu-latest
+    steps:
+      - id: set_variables
+        run: |
+          # make the repository lowercase
+          echo "repository=${GITHUB_REPOSITORY,,}" >> "$GITHUB_OUTPUT"
+          # truncate sha to 11 characters
+          echo "sha=${GITHUB_SHA::11}" >> "$GITHUB_OUTPUT"
+        env:
+          GITHUB_REPOSITORY: ${{ github.repository }}
+          GITHUB_SHA: ${{ github.sha }}
+      - name: Draft a release
+        id: draft_a_release
+        uses: release-drafter/release-drafter@v5
+        with:
+          config-name: release-drafter.yml
+          publish: true
+          prerelease: true
+          tag: "$RESOLVED_VERSION-beta"
+          name: "$RESOLVED_VERSION-beta"
+          header: |
+            ## This is a pre-release version of IQEngine
+            
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Checkout the repository
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN  }}
+      - name: Build and Push Docker Image
+        uses: docker/build-push-action@v2
+        if: steps.draft_a_release.outputs.tag_name != ''
+        with:
+          labels: |
+            org.opencontainers.image.source=https://github.com/IQEngine/IQEngine
+          push: true
+          tags: |
+            ghcr.io/${{ steps.set_variables.outputs.repository }}:${{ steps.set_variables.outputs.sha }}
+            ghcr.io/${{ steps.set_variables.outputs.repository }}:${{ steps.draft_a_release.outputs.tag_name }}
+            ghcr.io/${{ steps.set_variables.outputs.repository }}:beta
```

Comment:
Will you not get an already exists error with this tag?

Reply:
Whe you push a container with the same tag as one that already exist it just override this tag pointing to the new container image you have.